lib/assetmanager.js

```diff
@@ -485,8 +516,9 @@
             this._fileManager.updateBasePath(this._document);
         }
 
-        if (change.resolution || change.bounds) {
-            this._reset(true);
+        if (change.resolution || change.bounds || change.generatorSettings) {
```

Comment:
Is there only one change per event or can you ever get one that has both a bounds change and a generatorSettings change?